csharp/src/Ice/Proxy.cs

```diff
@@ -437,37 +457,12 @@ public static class Proxy
                             }
                         }
 
-                        if (retryPolicy.Retryable == Retryable.No)
+                        if (releaseRequestAfterSent || retryPolicy.Retryable == Retryable.No)
```

Comment:
If `releaseRequestAfterSent` is true but the request isn't sent... can't we still retry? Won't this check prevent such a retry?

Reply:
good catch, we should check for sent here to allow the retry

Reply:
fixed